synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -0,0 +1,295 @@
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include <ETL/handle>
+#include <gui/app.h>
+#include <gui/dialogs/canvasproperties.h>
+#include <gui/localization.h>
+#include <gui/resourcehelper.h>
+#include <gui/widgets/widget_link.h>
+#include <synfig/renddesc.h>
+#include <synfigapp/canvasinterface.h>
+
+#include "canvasresize.h"
+
+#endif
+
+using namespace etl;
+using namespace Gtk;
+using namespace sigc;
+using namespace std;
+using namespace studio;
+using namespace synfig;
+using namespace synfigapp;
+
+CanvasResize::CanvasResize(
+		Window &parent, handle<CanvasInterface> &ci,
+		CanvasProperties &cp, bool ratio_wh_toggle_status):
+	Dialog            ("", parent),
+	canvas_interface  (ci),
+	rend_desc         (ci->get_canvas()->rend_desc()),
+	canvas_properties (cp),
+	builder           (Builder::create_from_file(ResourceHelper::get_ui_path("canvas_resize.glade"))),
+	grid_content      (nullptr),
+	width             (nullptr),
+	height            (nullptr),
+	toggle_ratio_wh   (nullptr),
+	IS_TOGGLED_DEFAULT(ratio_wh_toggle_status),
+	is_toggled_default(true),
+	update_lock       (0)
+{
+	set_up_builder_widgets();
+	set_up_toggle_tooltips();
+
+	// Add builder grid content to this dialog window content
+	get_content_area()->add(*grid_content);
+
+	set_up_action_widgets();
+
+	set_resizable(false);
+
+	// Decide which dialog this is: "Resize with Content" or "Set Size"
+	set_up_dialog_label();
+	IS_TOGGLED_DEFAULT ? set_image_flags() : set_canvas_flags();
+
+	refresh_title();
+	refresh_content();
+
+	// Finally, connect all our signals
+	set_up_signals();
+}
+
+void CanvasResize::set_up_builder_widgets()
+{
+	builder->get_widget("grid_content", grid_content);
+	builder->get_widget("width"       , width);
+	builder->get_widget("height"      , height);
+	builder->get_widget("label"       , label);
+	builder->get_widget_derived("toggle_ratio_wh", toggle_ratio_wh);
+}
+
+void CanvasResize::set_up_toggle_tooltips()
+{
+	toggle_ratio_wh->property_tooltip_active().set_value(_("Unlink width and height"));
+	toggle_ratio_wh->property_tooltip_inactive().set_value(_("Link width and height"));
+
+	// Force refresh new tooltip messages
+	toggle_ratio_wh->toggled();
+}
```

Comment:
@rodolforg Since this class is isolated I bet it's safe to change the custom property tooltip message?

Reply:
Did you mean force it in code?
I think it's fine.
Btw, you don't need to use set_value()
You can use = :)